### DISCUSSION TITLE
Automatic restart if health check fails

### DISCUSSION THREAD

**Discussion by bcrooker** [teslamate-org/teslamate#1378] [Show and tell]
While Teslamate has been pretty stable (I only had 1 unhealthy state prior to this weekends Tesla auth issues), I wanted to create a script to detect this and react to it.  I wrote this simple script.  I set the script up to run via cron.  The "teslamate/restart" line is an HTTP IN node-red flow that triggers an email to let me know that a restart was attempted.

```
#!/bin/bash

if curl -s "http://localhost:4000" | grep "Health check failed"
then
    curl -s "http://192.168.1.11:1880/teslamate/restart"

    sudo docker-compose restart
else
fi
```

**Comment by LelandSindt** [comment]
I don't recall the last time I had to restart TeslaMate to clear a "Health check failed" status... When I notice my [Gaussmeter](https://github.com/gaussmeter/gaussmeter) flash red I can usually clear it by waking my car, or letting it clear on its own.

... how often do you see `Health check failed` that does not clear on its own?

**Reply by rmyadsk** [reply]
Yes, usually it is quite resilient. When Tesla abruptly changed their WAF configuration last week (#1369) this was a widespread issue for several users until [this commit](https://github.com/adriankumpf/teslamate/commit/1384023bf5d3a9c13aeea4856f78b5b802f644ad) resolved it.
